atc/engine/builder/delegate_factory.go

```diff
@@ -506,6 +506,17 @@ func (delegate *buildStepDelegate) Finished(logger lager.Logger, succeeded bool)
 	logger.Info("finished")
 }
 
+func (d *buildStepDelegate) SelectedWorker(logger lager.Logger, workerName string) {
+	err := d.build.SaveEvent(event.SelectedWorker{
+		Time:       time.Now().Unix(),
```

Comment:
In order to associate this event to its step in the UI, I think we'll need to add an `Origin` field.

Side note: ideally this would just be a field for the step's ID, since there's no stdout/stderr, but it looks like we just use `Origin` and leave the source blank in the `Initialize`/`Start`/`Finish` events, so I guess we should do the same here.